lib/CMM/ui/my_cmm_tab/add_cmm_page.dart/add_cmm_page.dart

```diff
@@ -0,0 +1,121 @@
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:heroicons/heroicons.dart';
+import 'package:image_picker/image_picker.dart';
+import 'package:myecl/CMM/ui/cmm.dart';
+import 'package:myecl/CMM/ui/components/button.dart';
+import 'package:myecl/tools/functions.dart';
+import 'package:myecl/tools/ui/widgets/image_picker_on_tap.dart';
+import 'package:qlevar_router/qlevar_router.dart';
+
+class AddCMMPage extends HookWidget {
+  const AddCMMPage({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final poster = useState<Uint8List?>(null);
+    final posterFile = useState<Image?>(null);
+    final ImagePicker picker = ImagePicker();
+
+    void displayAdvertToastWithContext(TypeMsg type, String msg) {
+      displayToast(context, type, msg);
+    }
+
+    return CMMTemplate(
+      child: Column(
+        children: [
+          const Spacer(),
+          FormField<File>(
+            validator: (e) {
+              if (poster.value == null) {
+                return "Choix poster";
+              }
+              return null;
+            },
+            builder: (formFieldState) => Center(
+              child: Stack(
+                clipBehavior: Clip.none,
+                children: [
+                  ImagePickerOnTap(
+                    picker: picker,
+                    imageBytesNotifier: poster,
+                    imageNotifier: posterFile,
+                    displayToastWithContext: displayAdvertToastWithContext,
+                    child: Container(
+                      decoration: BoxDecoration(
+                        boxShadow: [
+                          BoxShadow(
+                            color: formFieldState.hasError
+                                ? Colors.red
+                                : Colors.black.withOpacity(0.1),
+                            spreadRadius: 5,
+                            blurRadius: 10,
+                            offset: const Offset(2, 3),
+                          ),
+                        ],
+                      ),
+                      child: posterFile.value != null
+                          ?
+                          // ? Container(
+                          //     decoration: BoxDecoration(
```

Comment:
Just don't forget to delete it if you are certain not to use it anymore, it should not stay in the PR in the end